lektor/markdown.py

```diff
@@ -56,7 +67,9 @@ def make_markdown(env):
     env.plugin_controller.emit("markdown-config", config=cfg)
     renderer = cfg.make_renderer()
     env.plugin_controller.emit("markdown-lexer-config", config=cfg, renderer=renderer)
-    return mistune.Markdown(renderer, **cfg.options)
+    if _old_mistune:
+        return mistune.Markdown(renderer, **cfg.options)
+    return mistune.Markdown(renderer)
```

Comment:
Mistune v1.x appears to support plugins.  Those would (I think) be passed to the `Markdown` constructor.
I'm not familiar enough to say, but would there be worth in adding the hooks to `MarkdownConfig` so that Lektor plugins can configure the use of Mistune plugins?

Reply:
Thank you for the general review, it is very useful and makes the PR more maintainable.

I am also not super familiar with these particular bits, but you are right to point out that we were ignoring configuration data.
I found this bit of documentation (might be new, or maybe I'm fresh and noticed it today :-D) https://mistune.readthedocs.io/en/latest/api.html#mistune.create_markdown

Which seems somewhat compatible with the way the previous version was being used: https://mistune.readthedocs.io/en/v0.8.4/#options